### PR TITLE
Add uniform, automated linting

### PR DESCRIPTION
**Pull request by mattwthompson**
I find it difficult to read some changesets in this project, owing largely to things like long lines and inconsistent text wrapping. Enter (standard set of tools). The https://pre-commit.ci/ would make this a little smoother but is not necessary.

The changes here are mostly automated, though I had to make some manual fixes and I've also nested a few questions inside the code itself.

Happy to take responsibility for rebasing other PRs.

### CODE REVIEW

**Comment by mikemhenry** [comment]
I will merge this in once I get the other PRs merged in, run the pre-commit on the whole code base, squash in the commits, then add a file so the web-ui knows to skip that commit in the blame

**Comment by mikemhenry** [comment]
just updated the oe license

**Comment by mattwthompson** [comment]
Merging upstream changing in now ...

**Review by mikemhenry — APPROVED** [review]

**Pull request merged**
